qiskit/quantum_info/operators/mixins/linear.py

```diff
@@ -38,13 +38,29 @@ class LinearMixin(MultiplyMixin, ABC):
     """
 
     def __add__(self, other):
+        # enable easy use of sum(...)
+        if other == 0:
+            return self
+
+        qargs = getattr(other, "qargs", None)
+        return self._add(other, qargs=qargs)
+
+    def __radd__(self, other):
+        # enable easy use of sum(...)
+        if other == 0:
+            return self
+
         qargs = getattr(other, "qargs", None)
         return self._add(other, qargs=qargs)
 
     def __sub__(self, other):
         qargs = getattr(other, "qargs", None)
         return self._add(-other, qargs=qargs)
 
+    def __rsub__(self, other):
+        qargs = getattr(other, "qargs", None)
+        return -self._add(other, qargs=qargs)
```

Comment:
Did you mean `(-self)._add`? It'd be good to add tests of the new from-the-right magic methods, but other than your scalar 0, I'm not sure they'd ever be successfully called?

Reply:
The right-side magic methods would get called when adding two custom classes, where only one of them might support the addition with the other one. I could create a custom class for testing if you want. Or I test it using the magic `0` case again...

Reply:
I pushed some updates but it is still not ideal in terms of testing. Let me know what you think..